spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jdbc/EmbeddedDatabaseConnection.java

```diff
@@ -122,10 +123,23 @@ public String getUrl(String databaseName) {
 	 * @param driverClass the driver class
 	 * @return true if the driver class is one of the embedded types
 	 */
+	@Deprecated
 	public static boolean isEmbedded(String driverClass) {
 		return driverClass != null && (matches(HSQL, driverClass) || matches(H2, driverClass)
 				|| matches(DERBY, driverClass) || matches(HSQLDB, driverClass));
 	}
+	/**
+	 * Convenience method to determine if a given driver class name and url represents an embedded
+	 * database type.The exception is made for the H2 database for embedded types.
+	 * @param driverClass the driver class
+	 * @param url the jdbc url
+	 * @return true if the driver class is one of the embedded types
+	 */
+	public static boolean isEmbedded(String driverClass, String url)
+	{
+		return (driverClass != null && (matches(HSQL, driverClass) || ( matches(H2, driverClass) && !StringUtils.hasText("mem"))
```

Comment:
The updated code does not use the `url` argument. This also indicates that tests to validate the change of behaviour are missing.

The check on the url must be more narrowed, something like `:h2:mem:` perhaps?